### PR TITLE
[finishes 108291350] Ability to make a given screen the default screen

### PR DESCRIPTION
**Pull request by mehalshah**

### CODE REVIEW

File: apps/src/applab/designMode.js [hunk 1]
```diff
@@ -308,6 +308,23 @@ designMode.updateProperty = function(element, name, value) {
...
+        // Make all other screens not default
+        var allScreens = document.querySelectorAll('#designModeViz [class=screen]');
+        Array.prototype.forEach.call(allScreens, function (item) {
+          if (item.id !== 'design_' + element.id && item.id !== element.id) {
+            item.setAttribute('is-default', 'false');
+          }
+        });
+
+        // Change the order in the dropdown so the new default goes to the top
+        console.log('changing screen order');
```
Comment (Bjvanminnen) [inline apps/src/applab/designMode.js:325]: We try to keep console.log statements out of our production code.

File: apps/src/applab/designMode.js [hunk 2]
```diff
@@ -778,9 +795,12 @@ designMode.getCurrentScreenId = function() {
...
  */
 designMode.loadDefaultScreen = function () {
+  var potentialDefaultScreens = $('#designModeViz [class=screen][is-default=true]');
   var defaultScreen;
   if ($('#designModeViz .screen').length === 0) {
     defaultScreen = designMode.createScreen();
+  } else if (potentialDefaultScreens.length === 1) {
+    defaultScreen = elementUtils.getId(potentialDefaultScreens[0]);
   } else {
     defaultScreen = elementUtils.getId($('#designModeViz .screen')[0]);
```
Comment (Bjvanminnen) [inline apps/src/applab/designMode.js]: With this design, we should never hit this last else, right?

File: dashboard/test/ui/features/eyes.feature [hunk 3]
```diff
@@ -161,10 +161,11 @@ Scenario:
   Then I see no difference for "evaluated expression"
   And I close my eyes
 
-@dashboard_db_access
```
Comment (Bjvanminnen) [inline dashboard/test/ui/features/eyes.feature:164]: Should we be using @as_student here?

File: apps/src/applab/designMode.js [hunk 4]
```diff
@@ -308,6 +308,23 @@ designMode.updateProperty = function(element, name, value) {
       break;
     case 'readonly':
       element.setAttribute('contenteditable', !value);
+    case 'is-default':
+      element.setAttribute('is-default', value);
```
Comment (Bjvanminnen) [inline apps/src/applab/designMode.js:312]: I think we want this to be data-is-default http://stackoverflow.com/a/1735268/2506748

**Comment by Bjvanminnen** [comment]
Regarding your comment from the bug "I don't want the DOM ordering to have any meaning as far as what screen loads first, or what is first in the dropdown - it's not intuitive to me that the first DOM element is necessarily the default. I'm taking a longer way of doing this but it'll have the same effect."

Right now, it seems like all we've done is introduce a second way of determining the default screen. When I create a new app (or load an app that was created before this PR), we use DOM order to figure out which screen is "default". Then, once I check one of the screens as default, we start using this new attribute based method.

If we want to go with this approach, we should probably do two things (a) when a new app is created, the first screen gets marked as default using the attribute approach (b) when we load an app with no default screen attributes, we mark the first screen in the DOM ordering as default. We're still using both approaches in (b), but only as a sort of migration type step.

That being said, my own feeling is that it would be simpler to just always depend on DOM order, and whenever we change the default screen, shuffle the DOM.

One other thing I just thought of - what does/should happen when I try to unmark a screen as default? It's not clear to me that one should be able to do this. I think this is part of why I was originally thinking button instead of checkbox.

File: apps/src/applab/designMode.js (hunk 1 above)
Comment (mehalshah) [inline apps/src/applab/designMode.js:325]: Whoops - will remove

File: dashboard/test/ui/features/eyes.feature (hunk 3 above)
Comment (mehalshah) [inline dashboard/test/ui/features/eyes.feature:164]: Should be part of a different change

File: apps/src/applab/designMode.js (hunk 4 above)
Comment (mehalshah) [inline apps/src/applab/designMode.js:312]: Will change

**Comment by mehalshah** [comment]
Let me consider your comment for a bit.

**Pull request closed without merging**